pelican/settings.py

```diff
@@ -156,79 +162,67 @@ def load_source(name, path):
 
 
 def read_settings(path=None, override=None):
+    settings = override or {}
+
     if path:
-        local_settings = get_settings_from_file(path)
-        # Make the paths relative to the settings file
+        settings = dict(get_settings_from_file(path), **settings)
+
+    if settings:
+        settings = handle_deprecated_settings(settings)
```

Comment:
This probably belongs to `configure_settings`.

Reply:
There was already some deprecation handling here, directly in `read_settings`. This unites all deprecation handling in a dedicated method, which I feel improves maintainability. Handling of deprecated settings that involve paths needs to occur before making relative paths absolute, so by extension before `configure_settings`.

Reply:
Oh, I definitely agree with moving deprecation to its own function. No argument there. I guess it's fine. I was just suggesting moving the function call in `configure_settings`, but yeah, to do that `make absolute` bit also need to be moved I guess. I guess this is fine :).